pysnooper/utils.py

```diff
@@ -58,14 +55,16 @@ def get_repr_function(item, custom_repr):
     return repr
 
 
-def get_shortish_repr(item, custom_repr=()):
+def get_shortish_repr(item, custom_repr=(),
+                      need_truncate=True, max_length=100):
     repr_function = get_repr_function(item, custom_repr)
     try:
         r = repr_function(item)
     except Exception:
         r = 'REPR FAILED'
     r = r.replace('\r', '').replace('\n', '')
-    r = truncate(r, MAX_VARIABLE_LENGTH)
```

Comment:
I'm noticing a potential problem that's related to this PR but not caused by it. If someone provides a custom repr, it can still get truncated. I think a custom repr should override that and be shown in full.

Reply:
Yes. I've noticed that too.
However, user could pass "need_truncate=False" when he wants to use custom repr without truncated.
This option is useful for user to decide what he wants.